reactor-core/src/main/java/reactor/core/publisher/ContextPropagation.java

```diff
@@ -100,6 +106,25 @@ public static Function<Context, Context> contextCapture(Predicate<Object> captur
 		return new ContextCaptureFunction(captureKeyPredicate, null);
 	}
 
+	public static <T, R> BiConsumer<T, SynchronousSink<R>> contextRestoreForHandle(BiConsumer<T, SynchronousSink<R>> handler, CoreSubscriber<? super R> actual) {
+		if (!ContextPropagation.isContextPropagationAvailable() || !actual.currentContext().hasKey(ContextPropagation.CAPTURED_CONTEXT_MARKER)) {
+			return handler;
+		}
+		return (t, s) -> {
+			try (ContextSnapshot.Scope ignored = ContextSnapshot.setThreadLocalsFrom(actual.currentContext())) {
```

Comment:
this is untested and doesn't work with anything other than `ContextRegistry.getInstance()`